apps/devhub/tests/test_utils.py

```diff
@@ -0,0 +1,373 @@
+
+        message = {'id': ('a', 'b', 'c'),
+                   'signing_severity': 'low',
+                   'context': ('x', 'y', 'z'),
+                   'context_data': {'function': 'foo_bar'},
+                   'file': 'foo.js'}
+
+        self.compare(message, {'id': ('d', 'e', 'f')}, {})
+        self.compare(message, {'signing_severity': 'high'}, {})
+        self.compare(message, {'context': ('a', 'b', 'c')}, {})
+        self.compare(message, {'context_data': {}}, {})
+        self.compare(message, {'file': 'these-are-not-the-droids.js'}, {})
+
+        # Token non-signing message.
+        self.compare({'id': ('a', 'b', 'c'),
+                      'context': ('x', 'y', 'z'),
+                      'file': 'foo.js'},
+                     {'id': ()}, {})
+
+
+class TestValidationAnnotatorBase(amo.tests.TestCase):
+
+    def setUp(self):
+        # FIXME: Switch to factories.
+        # self.file = FileFactory(version__version='1.0',
+        #                         version__addon__guid='test-desktop@nowhere')
+        # self.file_1_1 = FileFactory(version__version='1.1',
+        #                             version__addon=self.file.version.addon)
+
+        # Create File objects for version 1.0 and 1.1.
+        self.addon = Addon.objects.create(guid='test-desktop@nowhere',
+                                          slug='test-amo-addon',
+                                          is_listed=False,
+                                          status=amo.STATUS_PUBLIC)
+
+        self.version = Version.objects.create(version='1.0', addon=self.addon)
+        self.file = File.objects.create(filename='desktop.xpi',
+                                        version=self.version,
+                                        status=amo.STATUS_PUBLIC)
+
+        self.version_1_1 = Version.objects.create(version='1.1',
+                                                  addon=self.addon)
+        self.file_1_1 = File.objects.create(filename='desktop.xpi',
+                                            version=self.version_1_1)
+
+        # Create a FileUpload object for an XPI containing version 1.1.
+        path = os.path.join(settings.ROOT,
+                            'apps/devhub/tests/addons/desktop.xpi')
+        self.file_upload = FileUpload.objects.create(path=path)
+        self.xpi_version = '1.1'
+
+        # Patch validation tasks that we expect the annotator to call.
+        self.patchers = []
+        self.save_file = self.patch(
+            'devhub.tasks.handle_file_validation_result').subtask
+        self.save_upload = self.patch(
+            'devhub.tasks.handle_upload_validation_result').subtask
+
+        self.validate_file = self.patch(
+            'devhub.tasks.validate_file').subtask
+        self.validate_upload = self.patch(
+            'devhub.tasks.validate_upload').subtask
+
+    def tearDown(self):
+        for patcher in self.patchers:
+            patcher.stop()
```

Comment:
Do we need a http://www.voidspace.org.uk/python/mock/mock.html?highlight=reset#mock.Mock.reset_mock here?